webroot/js/local.js

```diff
@@ -1,58 +1,65 @@
 jQuery(document).on('ready', function() {
 
-	jQuery('.checkbox input').iCheck({
-		checkboxClass: 'icheckbox_flat',
-		increaseArea: '20%'
-	});
-
-	jQuery('input.autocomplete').each(function() {
-		var _this = jQuery(this);
-		var cache = {};
-		var url = _this.data('url');
-		var update = _this.data('linked-to');
-
-		if (update) {
-			update = $('#' + update);
-		}
-
-		_this.autocomplete({
-			minLength: 0,
-
-			select: function(event, ui) {
-				console.log(ui.item);
-				_this.val(ui.item ? ui.item.value : this.value);
-
-				if (update) {
-					update.val(ui.item ? ui.item.id : '');
-				}
-			},
-
-			source: function(request, response) {
-				var term = request.term;
-
-				if (cache.hasOwnProperty(term)) {
-					response(cache[term]);
-					return;
-				}
-
-				$.getJSON(url, request, function(data, status, xhr) {
-					var result = [];
-
-					jQuery.each(data.data, function(key, value) {
-						console.log(key, value);
-						result.push({"id": key, "value": value });
-					});
-
-					console.log(result);
-					cache[term] = result;
-					response(result);
-				});
-			}
-		})
-		.focus(function() {
-			_this.autocomplete('search');
-		});
-
-	});
+  var bulkActionForm = jQuery('.bulk-actions');
+  if (bulkActionForm.length) {
+    bulkActionForm.submit(function (e) {
+      var action = jQuery('.bulk-actions .bulk-action-submit select').val();
+      if (!action) {
+        return e.preventDefault();
+      }
+
+      bulkActionForm.attr('action', action);
+    });
+  }
+
+  jQuery('input.autocomplete').each(function () {
+    var _this = jQuery(this);
+    var cache = {};
+    var url = _this.data('url');
+    var update = _this.data('linked-to');
+
+    if (update) {
+      update = $('#' + update);
+    }
+
+    _this.autocomplete({
+      minLength: 0,
+
+      select: function(event, ui) {
+        console.log(ui.item);
+        _this.val(ui.item ? ui.item.value : this.value);
+
+        if (update) {
+          update.val(ui.item ? ui.item.id : '');
+        }
+      },
+
+      source: function(request, response) {
+        var term = request.term;
+
+        if (cache.hasOwnProperty(term)) {
+          response(cache[term]);
+          return;
+        }
+
+        $.getJSON(url, request, function(data, status, xhr) {
+          var result = [];
+
+          jQuery.each(data.data, function(key, value) {
+            console.log(key, value);
+            result.push({"id": key, "value": value });
+          });
+
+          console.log(result);
+          cache[term] = result;
+          response(result);
+        });
+      }
+    })
+    .focus(function() {
+      _this.autocomplete('search');
+    });
+
+  });
```

Comment:
I restrict to individual element types, not to "pages", so that people can extend and as long as they follow those specific selectors, they'll be fine. For now, I don't think it's worth a micro-optimization to split it into action-specific js files.

Doing it that way, we'd have something like 4 dom-ready calls on a page for no good reason ;)